pkgs/test_api/lib/src/frontend/stream_matcher.dart

```diff
@@ -179,7 +179,7 @@ class _StreamMatcher extends AsyncMatcher implements StreamMatcher {
       buffer.writeln(indent(eventsString, first: 'emitted '));
       if (result.isNotEmpty) buffer.writeln(indent(result, first: '  which '));
       return buffer.toString().trimRight();
-    }, onError: (error) {
+    }, onError: (Object error) {
```

Comment:
this is weird. You'd think `Object` would be inferred here!

Reply:
The `onError` argument has a type of `Function`. In the _ideal_ world we'd have union types so we could type it as `void Function(Object)|void Function(Object, StackTrace)`. For now we don't get _any_ inference for argument types, and not static checking for argument count.

Reply:
yup